src/test/java/io/reactivex/internal/operators/observable/ObservableBufferTest.java

```diff
@@ -2136,4 +2165,15 @@ public void bufferSkipDoubleOnSubscribe() {
             }
         });
     }
+
+    @Test
+    public void bufferExactFailingSupplier() {
+        Scheduler.Worker w = new TestScheduler().createWorker();
```

Comment:
The list supplier should fail, calling onError and onCompleted this way is just a plain protocol violation.

Do something this instead:

```java
Observable.empty()
.buffer(10, 10, TimeUnit.SECONDS, Schedulers.computation(), new Callable<List<Object>>() {
    @Override public void call() {
        throw new TestException();
    }
})
.test()
.awaitDone(5, TimeUnit.SECONDS)
.assertFailure(TestException.class)
;
```